File: login_simulator.js
const correctUser = {
    username: "Mike",
    password: "c0mp14x_password",
    twoFA: "27892",
    securityAnswer: "Fluffy"
};

let failedAttempts = 0;
let isLocked = false;

const form = document.getElementById("loginForm");
const usernameInput = document.getElementById("username");
const passwordInput = document.getElementById("password");
const twoFAInput = document.getElementById("twoFA");
const securityQuestionDiv = document.getElementById("securityQuestion");
const securityAnswerInput = document.getElementById("securityAnswer");
const messageDiv = document.getElementById("message");

function updateSecurityLayers() {
    if (failedAttempts >= 1) {
        twoFAInput.classList.remove("hidden");
    } else {
        twoFAInput.classList.add("hidden");
        twoFAInput.value = "";
    }

    if (failedAttempts >= 2) {
        securityQuestionDiv.classList.remove("hidden");
        securityAnswerInput.classList.remove("hidden");
    } else {
        securityQuestionDiv.classList.add("hidden");
        securityAnswerInput.classList.add("hidden");
        securityAnswerInput.value = "";
    }
}

function failLogin(text) {
    failedAttempts++;
    messageDiv.textContent = `${text} Failed attempts: ${failedAttempts}/3`;
    messageDiv.style.color = "#E86252";

    updateSecurityLayers();

    if (failedAttempts >= 3) {
        isLocked = true;
        messageDiv.textContent = "Too many failed attempts. Account locked.";
        form.querySelector("button").disabled = true;

        setTimeout(() => {
            window.location.href = "game.html?mode=hard";
        }, 2500);
    }
}

form.addEventListener("submit", function(event) {
    event.preventDefault();

    if (isLocked) {
        messageDiv.textContent = "Account is locked. Please contact support.";
        messageDiv.style.color = "#E86252";
        return;
    }

    const username = usernameInput.value.trim();
    const password = passwordInput.value;
    const twoFA = twoFAInput.value.trim();
    const securityAnswer = securityAnswerInput.value.trim();

    if (username !== correctUser.username) {
        failLogin("Incorrect username.");
        return;
    }

    if (password !== correctUser.password) {
        failLogin("Incorrect password.");
        return;
    }

    if (failedAttempts >= 1 && twoFA !== correctUser.twoFA) {
        failLogin("Incorrect 2FA code.");
        return;
    }

    if (failedAttempts >= 2 && securityAnswer !== correctUser.securityAnswer) {
        failLogin("Incorrect security answer.");
        return;
    }

    messageDiv.textContent = "Login successful! Redirecting...";
    messageDiv.style.color = "#871F78";

    setTimeout(() => {
        window.location.href = "game.html?mode=easy";
    }, 1500);
});

updateSecurityLayers();

File: login_simulator.cpp
#include <iostream>
#include <string>
#include <unistd.h>

using namespace std;

string gethiddenpassword() {
    char* pwd = getpass("Enter password: ");
    return string(pwd);
}

string gethidden2facode() {
    char* code = getpass("Enter 2FA code: ");
    return string(code);
}

struct User {
    string username;
    string password;
    string twoFAcode;
    string securityQuestion;
    string securityAnswer;
    int failedLoginAttempts;
    bool isLocked;
};

void handleFailure(User& user, const string& message) {
    user.failedLoginAttempts++;
    cout << message << "\n";
    cout << "Failed login attempts: " << user.failedLoginAttempts << "\n";
    if (user.failedLoginAttempts >= 3) {
        user.isLocked = true;
        cout << "Too many failed login attempts. Account is locked.\n";
    }
} 

bool userLocked(User& user) {
    if (user.isLocked) {
        cout << "Your account is locked. Please contact support.\n";
        exit(0);
    }
    return false;
}

bool usernameCorrect = false;
bool passwordCorrect = false;
bool twoFACorrect = false;

int main() {
    User user = {
        "Mia", 
        "c0mp14x_password", 
        "27892", 
        "What is the name of your first pet?", 
        "Fluffy", 
        0, 
        false
    };

    while(!user.isLocked) {
        string username;
        cout << "Username: ";
        cin >> username;

        if (username != user.username) {
            handleFailure(user, "Incorrect username.");
            continue;
        }
        string password = gethiddenpassword();
        if (password != user.password) {
            handleFailure(user, "Incorrect password.");
            continue;
    }
        if(user.failedLoginAttempts >= 1){
            string code = gethidden2facode();
            if (code != user.twoFAcode) {
                handleFailure(user, "Incorrect 2FA code.");
                continue;
        }
    }

        if(user.failedLoginAttempts >= 2){
            string answer;
            cout << user.securityQuestion << " ";
            cin >> answer;
            if (answer != user.securityAnswer) {
                handleFailure(user, "Incorrect security answer.");
                continue;
        }
    }
        cout << "Login successful! Welcome, " << user.username << ".\n";
        return 0;
    }
    cout << "Your account is locked. Please contact support.\n";
    return 0;
}
    
